### PR TITLE
make Velero uninstall backward compatible

### PR DESCRIPTION
**Pull request by qiuming-best**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)
When using the newer Velero  CLI (v1.12) to uninstall the old Velero  CLI deployed environment, there could be some backward-compatible problems.

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/cmd/cli/uninstall/uninstall.go [hunk 1]
```diff
@@ -348,23 +354,19 @@ func gracefullyDeleteResource(ctx context.Context, kbClient kbclient.Client, nam
...
+			itemLen += reflect.ValueOf(resToDelete[i]).Elem().FieldByName("Items").Len()
+			if itemLen > 0 {
+				fmt.Print(".")
+				return false, nil
+			}
 		}
 
-		if len(restoreList.Items)+len(dataUploadList.Items)+len(dataDownloadList.Items) > 0 {
-			fmt.Print(".")
+		if itemLen > 0 {
```
Comment (ywk253100) [inline pkg/cmd/cli/uninstall/uninstall.go:369]: Seems this line is useless as the function always returns in line 365 if itemLen > 0?

Comment (qiuming-best) [reply pkg/cmd/cli/uninstall/uninstall.go:369]: Indeed it's useless, it's But If I do not add `itemLen > 0 `, the compile will failed with a `missing return`

Comment (ywk253100) [reply pkg/cmd/cli/uninstall/uninstall.go:369]: Why? You remove both code blocks of `if` and `else`? Shouldn't you just remove the `if` block and keep the last `return`?

Comment (qiuming-best) [reply pkg/cmd/cli/uninstall/uninstall.go:369]: it's reported by ConditionFunc in PollImmediate, ConditionFunc should have a return for every condition

**Review by ywk253100 — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Pull request merged**